eclipse-sirius/sirius-web
# [3382] Improve hide capabilities in the palette

Bug: https://github.com/eclipse-sirius/sirius-web/issues/3382

# Pull request template

## General purpose
What is the main goal of this pull request?

## Project management

## Architectural decision records (ADR)

## Dependencies

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- Hooks are properly typed:

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose

### Architecture

## Review

### How to test this PR?
1. Create a Papaya studio and a Papaya model with a Package containing several Classes.
2. In the Papaya Studio, delete the palette in Package and create a new one
3. The palette should contain a new tool section with hide/show tools
4. Click on the Package in the Papaya model, new hide/show tools should be visible (some are only visible if at least 1 child element is hidden)
